src/datasources/asset-calibration/AssetCalibrationDataSource.ts

```diff
@@ -50,35 +50,70 @@ export class AssetCalibrationDataSource extends DataSourceBase<AssetCalibrationQ
     super(instanceSettings, backendSrv, templateSrv);
   }
 
+  async runQuery(query: AssetCalibrationQuery, options: DataQueryRequest): Promise<DataFrameDTO> {
```

Comment:
Can you please write some tests for these changes?

Reply:
Added tests that verify the specific transformations, location query in our case, are created correctly.